clients/unshaded/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -450,13 +450,20 @@ private TachyonByteBuffer readLocalByteBuffer(int blockIndex, long offset, long
    *
    * @param blockInfo The blockInfo of the block to read.
    * @return TachyonByteBuffer containing the block.
+   * @throws IOException if the underlying
    */
-  TachyonByteBuffer readRemoteByteBuffer(ClientBlockInfo blockInfo) {
+  TachyonByteBuffer readRemoteByteBuffer(ClientBlockInfo blockInfo) throws IOException {
     // We call into the remote block in stream class to read a remote byte buffer
-    ByteBuffer buf =
-        RemoteBlockInStream.readRemoteByteBuffer(mTachyonFS, blockInfo, 0, blockInfo.length,
-            mTachyonConf);
-    return (buf == null) ? null : new TachyonByteBuffer(mTachyonFS, buf, blockInfo.blockId, -1);
+    // TODO: Avoid using this constructor, investigate the right way to read remote buffer here.
+    RemoteBlockInStream inStream = new RemoteBlockInStream(new TachyonFile(null, -1, null),
+        ReadType.NO_CACHE, -1, null, null, true);
```

Comment:
Does it make sense to make this into a static method in RemoteBlockInStream like `getDummyStream`?